### DISCUSSION TITLE
loadFromJSON v6

### DISCUSSION THREAD

**Discussion by brianp70x** [fabricjs/fabric.js#8059] [Q&A] [answered]
render does not seem to occur after loadFromJSON, objects are not visible until canvas is clicked .

using version v6 

Example @ [https://jsfiddle.net/brianp70/8xry0meb/22/](url)
buttons below canvas...
1st: click "stringify and clear"

2nd: click "Reload" method 1 or 2

**Comment by 17hz** [comment, marked as answer]
`loadFromJSON` has been  an Promise in v6, you should not  call `canvas.renderAll()`  as callback anymore

now you should use 
```js
canvas.loadFromJSON(json).then(function(){canvas.renderAll()})
```

**Reply by brianp70x** [reply]
does this also apply to loadSVGFromURL?

**Reply by mtrabelsi** [reply]
you can hover on it on vs code, it will show if it a promise or not...

**Reply by asturur** [reply]
All the functions have been moved to promise, we don't have callbacks anymore, or we shouldn't at least.

**Comment by Antonykibet** [comment]
Hello, why im i getting this error:
![image](https://github.com/user-attachments/assets/3b2c0190-7db3-48bc-a25e-566a815acd09)

**Reply by zhe-he** [reply]
Disable strict mode   OR
Destroy the canvas in useEffect
